Src/FluentAssertions/Common/Guard.cs

```diff
@@ -63,6 +63,22 @@ public static void ThrowIfArgumentIsEmpty<T>(IEnumerable<T> values, string param
         }
     }
 
+    public static void ThrowIfArgumentIsEmpty(string str, string paramName, string message)
+    {
+        if (str.Length == 0)
+        {
+            throw new ArgumentException(message, paramName);
+        }
+    }
+
+    public static void ThrowIfArgumentIsNegative(TimeSpan timeSpan, string paramName, string message)
+    {
+        if (timeSpan < TimeSpan.Zero)
```

Comment:
You might want to watch out for the special case of the "infinite timespan" here. IIRC it is set to something like "-1 ticks" or whatever (can use [`Timeout.InfiniteTimespan`](https://learn.microsoft.com/en-us/dotnet/api/system.threading.timeout.infinitetimespan?view=net-7.0) to get it).

That's usually considered a valid negative value even when you only want to deal with positive values.